clash-cores/test/Test/Cores/Xilinx/DcFifo.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Test.Cores.Xilinx.DcFifo ( tests ) where
+
+import qualified Prelude as P
+import qualified Data.List as L
+import Data.Maybe (catMaybes)
+import Control.Monad (replicateM)
+
+import Clash.Explicit.Prelude
+import Clash.Cores.Xilinx.DcFifo.Explicit
+
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Hedgehog (MonadGen, (===), property, forAll, Property)
+import Test.Tasty.Hedgehog (testPropertyNamed)
+import Test.Tasty (TestTree)
+
+tests :: TestTree
+tests = testPropertyNamed
+  "FIFO doesn't lose any data"
+  "prop_fifo"
+  prop_fifo
+
+intersperseStalls ::
+  -- | Data
+  [a] ->
+  -- | Stalls
+  [a] ->
+  [a]
+intersperseStalls (d:ds) s = d:s P.++ intersperseStalls ds s
+intersperseStalls [] _ = []
+
+prop_fifo :: Property
+prop_fifo = property $ do
+  xs <- fmap Just <$> forAll (replicateM 10 genData)
```

Comment:
As discussed on Slack (and for completeness here): it's better to start with:

```
n <- generate number of elements to produce bounded by [0, 10*size of fifo]
```

this makes sure you test overflow conditions.

Reply:
Ok, I can trigger an overflow! 

I will have separate tests in such cases I think.

Reply:
Right now the test I have is simple - it just tests that everything that is read out is in order (and present in the input) - is that enough?